okhttp/src/main/java/io/grpc/transport/okhttp/OkHttpClientTransport.java

```diff
@@ -263,18 +264,25 @@ public OkHttpClientStream newStream(MethodDescriptor<?, ?> method,
   @GuardedBy("lock")
   private void startStream(OkHttpClientStream stream) {
     Preconditions.checkState(stream.id() == null, "StreamId already assigned");
-    streams.put(nextStreamId, stream);
-    stream.start(nextStreamId);
+    int streamId = nextStreamId;
+    if (nextStreamId >= Integer.MAX_VALUE - 2) {
+      // Make sure nextStreamId is greater than all used ids, so that mayHaveCreatedStream()
```

Comment:
Add/enhance two tests:

1. Send data and half-close on a pending stream. You may optionally (in addition) remove the `mayHaveCreatedStream()` check in `OutboundFlowController`, but it is fine either way.
2. Add a test for receiving headers/data/trailers on stream with id MAX_VALUE-2